Legacy/bonej/src/main/java/org/bonej/plugins/ParticleAnalysis.java

```diff
@@ -516,6 +520,45 @@ public void filterParticles(final ImagePlus imp, final int[][] particleLabels, f
 		}
 		return eulerCharacters;
 	}
+	
+	/**
+	 * Calculate number of branches and total branch length by running 
+	 * Skeletonize3D and Analyze Skeleton on each particle.
+	 * 
+	 * Uses AnalyzeSkeleton's run() method with no arguments.
+	 * 
+	 * @param imp Input image
+	 * @param particleLabels particle label array
+	 * @param limits xyz limits of each particle
+	 * @param nParticles number of particles
+	 * @return array of AnalyzeSkelton result objects.
+	 * 	Use methods therein to get specific results out. Each result is a list of trees,
+	 * but there should be only one tree per particle so its result is at [0].
+	 */
+	static SkeletonResult[] getBranchLength(final ImagePlus imp,
+			final int[][] particleLabels, final int[][] limits, final int nParticles)
+	{
+		final SkeletonResult[] skeletonResults = new SkeletonResult[nParticles];
+		Thread[] threads = Multithreader.newThreads();
+		AtomicInteger ai = new AtomicInteger(1);
+		for (int thread = 0; thread < threads.length; thread++) {
+			threads[thread] = new Thread(() -> {
+				for (int i = ai.getAndIncrement(); i < nParticles; i = ai.getAndIncrement()) {
+					final ImagePlus particleImp = getBinaryParticle(i, imp, particleLabels,
+							limits, 1);
+					final PlugInFilter skeletoniser = new Skeletonize3D_();
+					skeletoniser.setup("", particleImp);
+					skeletoniser.run(null);
+					final AnalyzeSkeleton_ analyzeSkeleton_ = new AnalyzeSkeleton_();
```

Comment:
Each thread can create their `AnalyzeSkeleton_` and `Skeletonize_` instances once outside the loop

Reply:
Done.